intel_adsp: boot_complete must be done PRE_KERNEL_1

kv2019i opened the pull request:
This reverts commit 759e07bebe208f4cc1036a7525a80551e16a465c.

Zephyr boot fails on Intel Meteor Lake with this patch.

UPDATE: instead of revert, fix suggested by @nashif works:

intel_adsp: boot_complete must be done PRE_KERNEL_1

nashif commented:
Can we try

```diff
diff --git a/soc/xtensa/intel_adsp/common/boot_complete.c b/soc/xtensa/intel_adsp/common/boot_complete.c
index 791590e127..445fd57b06 100644
--- a/soc/xtensa/intel_adsp/common/boot_complete.c
+++ b/soc/xtensa/intel_adsp/common/boot_complete.c
@@ -29,4 +29,4 @@ int boot_complete(void)
        return 0;
 }

-SYS_INIT(boot_complete, EARLY, CONFIG_KERNEL_INIT_PRIORITY_DEVICE);
+SYS_INIT(boot_complete, PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEVICE);
```

instead?

kv2019i commented:
@nashif let me try that

kv2019i committed:
intel_adsp: boot_complete must be done PRE_KERNEL_1

Commit 759e07bebe20 ("intel_adsp: move memory window setup to
PRE_KERNEL_1") moved memory window setup from EARLY to
PRE_KERNEL_1. Similar change must be done to boot_complete, or
otherwise boot-up sequence will not be completed correctly
on all platforms.

Suggested-by: Anas Nashif <anas.nashif@intel.com>
Signed-off-by: Kai Vehmanen <kai.vehmanen@linux.intel.com>

nashif commented:
another option:

```diff
diff --git a/soc/xtensa/intel_adsp/common/boot.c b/soc/xtensa/intel_adsp/common/boot.c
index 75eed9d63b..5e47a26488 100644
--- a/soc/xtensa/intel_adsp/common/boot.c
+++ b/soc/xtensa/intel_adsp/common/boot.c
@@ -153,6 +153,7 @@ __imr void boot_core0(void)
        lp_sram_init();
        parse_manifest();
        sys_cache_data_flush_all();
+       mem_window_idle_exit();

        xtensa_vecbase_lock();
```

kv2019i commented:
The simple fix seems to hold up for me. Getting the boot_complete at unexpected time versus other events in the boot, seems to explain the boot fail. The python test harness used in one of the CI systems, seem to be fine with this, but rest of the test benches catch this as an error to boot up correctly. I'll work with @tmleman to test with more setups to be sure.

nashif commented:
> The simple fix seems to hold up for me. Getting the boot_complete at unexpected time versus other events in the boot, seems to explain the boot fail. The python test harness used in one of the CI systems, seem to be fine with this, but rest of the test benches catch this as an error to boot up correctly. I'll work with @tmleman to test with more setups to be sure.

what about the second option, which should be more or less the previous behaviour.

kv2019i commented:
@nashif wrote:
> ```diff
>  mem_window_idle_exit();
> ```

This didn't help. I think the problem is related to boot_complete(). i don't have logs working when the failure happens, but I'd guess the "(!device_is_ready(dev)" in boot_complete() is never true and Zephyr boot_complete() does not "win[0] = 5;" in boot_complete() leading to incomplete boot (as observed by the host -> Zephyr OS itself did boot up, it's just it didn't report the status correctly in SRAM).
